app/jitsi/models.py

```diff
@@ -38,38 +41,32 @@ def create(cls, username, avatar):
         return user
 
     @classmethod
-    def leave_room(cls, user_id, room_name):
-        room = Room.query.filter_by(name=room_name).first()
-        user_location = UserLocation.query.filter_by(user_id=user_id, room_id=room.id).first()
-        if user_location:
-            db.session.delete(user_location)
-            db.session.commit()
+    def leave_room(cls, user_id):
+        user = User.query.filter_by(id=user_id).first()
+        user.room = None
+        db.session.commit()
 
     @classmethod
     def enter_room(cls, user_id, room_name):
         room = Room.query.filter_by(name=room_name).first()
-
-        # Update location
-        user_location = UserLocation(user_id=user_id, room_id=room.id)
-        db.session.add(user_location)
+        user = User.query.filter_by(id=user_id).first()
+        user.room = room
 
         # Make room discovered by user
-        user_room_state = UserRoomState(user_id=user_id, room_id=room.id, discovered=True)
-        db.session.add(user_room_state)
+        user_room_state = UserRoomState.find_or_create(user_id=user_id, room_id=room.id)
+        user_room_state.discovered = True
```

Comment:
`UserRoomState.find_or_create` makes it so that we don't have duplicate UserRoomStates. This is also now enforced by a DB constraint ony `UserRoomState` that only allows one record per `user,room`.